GiniVision/Classes/ToolTipView.swift

```diff
@@ -0,0 +1,303 @@
+        ConstraintUtils.addActiveConstraint(item: tipContainer, attribute: .leading, relatedBy: .equal, toItem: textLabel, attribute: .leading, multiplier: 1, constant: -padding.left)
+        
+        // closeButton
+        ConstraintUtils.addActiveConstraint(item: closeButton, attribute: .width, relatedBy: .equal, toItem: nil, attribute: .notAnAttribute, multiplier: 1, constant: closeButtonWidth)
+        ConstraintUtils.addActiveConstraint(item: closeButton, attribute: .height, relatedBy: .equal, toItem: nil, attribute: .notAnAttribute, multiplier: 1, constant: closeButtonHeight)
+        ConstraintUtils.addActiveConstraint(item: closeButton, attribute: .centerY, relatedBy: .equal, toItem: tipContainer, attribute: .centerY, multiplier: 1, constant: 0)
+        ConstraintUtils.addActiveConstraint(item: closeButton, attribute: .leading, relatedBy: .equal, toItem: textLabel, attribute: .trailing, multiplier: 1, constant: itemSeparation)
+        ConstraintUtils.addActiveConstraint(item: tipContainer, attribute: .trailing, relatedBy: .equal, toItem: closeButton, attribute: .trailing, multiplier: 1, constant: padding.right)
+        
+        self.setNeedsLayout()
+    }
+    
+    // MARK: Draw arrow
+    class fileprivate func arrow(withHeight height:CGFloat, width:CGFloat, color:UIColor, position:ToolTipPosition) -> UIView {
+        let arrowView = UIView(frame: CGRect(x: 0, y: 0, width: width, height: height))
+        arrowView.backgroundColor = color
+        
+        let bezierPath = UIBezierPath()
+        
+        switch position {
+        case .above:
+            bezierPath.move(to: CGPoint(x: 0, y: 0))
+            bezierPath.addLine(to: CGPoint(x: width, y: 0))
+            bezierPath.addLine(to: CGPoint(x: width / 2, y: height))
+            bezierPath.addLine(to: CGPoint(x: 0, y: 0))
+        case .below:
+            bezierPath.move(to: CGPoint(x: 0, y: height))
+            bezierPath.addLine(to: CGPoint(x: width, y: height))
+            bezierPath.addLine(to: CGPoint(x: width / 2, y: 0))
+            bezierPath.addLine(to: CGPoint(x: 0, y: height))
+        case .left:
+            bezierPath.move(to: CGPoint(x: 0, y: 0))
+            bezierPath.addLine(to: CGPoint(x: height, y: width / 2))
+            bezierPath.addLine(to: CGPoint(x: 0, y: width))
+            bezierPath.addLine(to: CGPoint(x: 0, y: 0))
+        case .right:
+            bezierPath.move(to: CGPoint(x: height, y: 0))
+            bezierPath.addLine(to: CGPoint(x: 0, y: width / 2))
+            bezierPath.addLine(to: CGPoint(x: height, y: width))
+            bezierPath.addLine(to: CGPoint(x: height, y: 0))
+        }
+        bezierPath.close()
+        
+        let shapeLayer = CAShapeLayer()
+        shapeLayer.path = bezierPath.cgPath
+        arrowView.layer.mask = shapeLayer
+        return arrowView
+    }
+}
+
+// MARK: Show and hide tip methods
+
+extension ToolTipView {
+    
+    func show(animations:(() -> ())? = nil){
```

Comment:
The purpose of the `animations` block is not obvious. We should either rename it and add a parameter name to be invoked like this for ex. 
```
self.toolTipView?.show(withAuxiliaryAnimations: { // auxiliary just because it sounds fancier than additional :P
    blurEffect.alpha = 1
})
```

We could also extract the animation:
```
UIView.animate(withDuration: 0.5) {
    self.toolTipView?.show()
    blurEffect.alpha = 1
}
```

Reply:
You are right, it is not clear.